lib/x11test.pm

```diff
@@ -516,6 +516,36 @@ sub setup_mail_account {
     send_key_until_needlematch 'evolution_mail-max-window', 'super-up', 3, 3;
 }
 
+# Use AutoConfig file for firefox to predefine some user values
+# https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig
+sub prepare_firefox_autoconfig {
+    my ($self) = @_;
+    start_root_shell_in_xterm;
+
+    # Enable AutoConfig by pointing to a cfg file
+    type_string(q{cat <<EOF > $(rpm --eval %_libdir)/firefox/defaults/pref/autoconfig.js
+pref("general.config.filename", "firefox.cfg");
+pref("general.config.obscure_value", 0);
+EOF
+});
+    # Create AutoConfig cfg file
+    type_string(q{cat <<EOF > $(rpm --eval %_libdir)/firefox/firefox.cfg
+// Mandatory comment
+// https://firefox-source-docs.mozilla.org/browser/components/newtab/content-src/asrouter/docs/first-run.html
+pref("browser.aboutwelcome.enabled", false);
+pref("browser.startup.upgradeDialog.enabled", false);
+pref("privacy.restrict3rdpartystorage.rollout.enabledByDefault", false);
+EOF
+});
+
+    save_screenshot;
+
+    # Close root shell
+    send_key "ctrl-d", 1;
```

Comment:
Please avoid the positional parameter, use the syntax from https://github.com/os-autoinst/os-autoinst/blob/45a9ae5bd271e0415b8ef0835398ae0ffbac3e31/testapi.pm#L1343 or better don't use any "waiting" at all but explicit screen checks which in general can be faster and more stable, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/33e81b324e3079ca3ce3550578fdc7650e915569/tests/x11/sshxterm.pm#L32